hack/label_nodes/label_nodes.py

```diff
@@ -0,0 +1,48 @@
+#!/usr/bin/python3
+import sys
+import argparse
+import kubernetes.client
+import kubernetes.config
+
+JOBSET_NAME_KEY = "jobset.sigs.k8s.io/jobset-name"
+NODE_POOL_KEY = "cloud.google.com/gke-nodepool"
+
+def main(node_pools: str, jobset: str) -> None:
+    target_node_pools = set(node_pools.split(','))
+    print(f"Adding node label {JOBSET_NAME_KEY}={jobset} to the following node pools:")
+    for np in target_node_pools:
+        print(np)
+
+    input('Once confirmed, hit any key to continue, or ctrl+C to exit.')
+
+    # Load the Kubernetes configuration from the default location.
+    kubernetes.config.load_kube_config()
+
+    # Create a new Kubernetes client.
+    client = kubernetes.client.CoreV1Api()
+
+    # Get a list of all the nodes in the cluster.
+    nodes = client.list_node().items
+
+    body = {
+        "metadata": {
+            "labels": {
+                JOBSET_NAME_KEY: jobset,
```

Comment:
we need to label each node pool using the job name:

np1 gets the label value jobset-workers-slice1
np2 gets the label value jobset-workers-slice2
...

Reply:
done